src/engine/GroupBy.cpp

```diff
@@ -973,9 +1007,20 @@ std::vector<size_t> GroupBy::HashMapAggregationData::getHashEntries(
     hashEntries.push_back(iterator->second);
   }
 
-  for (auto& aggregation : aggregationData_)
-    std::visit([this](auto& arg) { arg.resize(getNumberOfGroups()); },
-               aggregation);
+  // TODO<C++23> use views::enumerate
+  auto idx = 0;
+  for (auto& aggregation : aggregationData_) {
+    const auto& aggregationTypeWithData = aggregateTypeWithData_.at(idx);
+    const auto numberOfGroups = getNumberOfGroups();
+    std::visit(
+        [&aggregationTypeWithData,
+         numberOfGroups]<SupportedAggregates T>(T& arg) {
+          ResizeVectorsVisitor<T>{}(arg, numberOfGroups,
+                                    aggregationTypeWithData);
+        },
+        aggregation);
```

Comment:
If it becomes short enough, mabye you can move your resizing lambda directly in here.
(depends on how short it can be).